### ISSUE TITLE
Please add a spell checker

### ISSUE THREAD

**Issue by ArthurZ** [RickStrahl/MarkdownMonster#64]
It would be nice to have an integrated spell checker.

Thanks!

**Comment by RickStrahl** [comment]
Come again? Spell checking is built-in via underlining misspelled text and selection.

![image](https://cloud.githubusercontent.com/assets/1374013/20852385/b2ac2efc-b889-11e6-98f4-32ed88e924b2.png)

**Comment by ArthurZ** [comment]
I ensured the SpellChecker was set to `True`.
Misspelling a word does not work like in your case:
![spellcheckerdoesntwork](https://cloud.githubusercontent.com/assets/701808/20852921/c45c4948-b8b6-11e6-8d93-acde824572f2.PNG)

**Comment by RickStrahl** [comment]
`True` is not a valid value. Valid value is `true`. 

You can click on the check icon in the Window header to toggle the spell checker.

**Comment by ArthurZ** [comment]
Yes, sorry, it was
`"EditorEnableSpellcheck": true,`
And toggling the mark above has no effect on anything.

**Comment by RickStrahl** [comment]
Not sure then.

What version of Windows and Internet Explorer are you running on? What language? And did you change dictionaries by any chance? Make sure dictionaries used exist in the `Editor` folder

**Comment by ArthurZ** [comment]
Win 7 Pro 64 Bit IE 11.
American English
The US dictionary exists among others:
http://screencast.com/t/wGSl8j7Ys

**Comment by RickStrahl** [comment]
Yeah I'm really not sure why it's not working. 

Can you do the following:

* Shut down MM
* Open `<program files>\MarkdownMonster\Editor\editorsettings.json`
* set `"isDebug": true,`
* save then run MM again

What if any error messages pop up on the bottom of the editor area? You're looking for errors that might relate to spell checking (some messages are OK).

When done reverse these steps.

**Comment by ArthurZ** [comment]
Whoa I do not have `editorsettings.json` Is that the problem?

**Comment by RickStrahl** [comment]
Sorry `editorsettings.js`... It's there.

**Comment by ArthurZ** [comment]
OK, did that. Edited to set it to true, started MM. 
Put in some garbled text and no messages, no spell checking. Nada.
Besides, I noticed, or it may by a deception, the SpellCheck toggle button is never becoming bright/active.

**Comment by RickStrahl** [comment]
No error messages on the bottom of the editor screen.

Sorry I'm out of ideas on why this wouldn't work especially if there are no errors. 

You can also rename your %appdata%\Markdown Monster\MarkdownMonster.json file and let a new one be created when you start up again. Shut down MM, rename the file, then restart (which should create a new file with all the stock defaults). Does that work any better?

The only other thing I can think of is, uninstall completely and resinstall to make sure all the right files are there. 

I haven't heard this from anybody else so this seems to be something specific to your setup, but I just can't think of what that could be.

**Comment by ArthurZ** [comment]
I saw MarkdownMonster.json in C:\Users\me\AppData\Roaming\Markdown Monster
Anyways deleted it. It re-appeared.
But did not heal the SpellChecker.
I will un-install on next MM version release.

**Comment by RickStrahl** [comment]
1.0.22 is available now.

To do a complete uninstall reinstall:

* Go to Programs Features
* Remove Markdown Monster
* Go to Program Files\Markdown Monster and delete anything left
* Go to %appdata%\Markdown Monster  and delete or rename this folder
* Install latest version

I have a feeling this won't help in your case, but worth a try.

One more thing I thought of here. It might be that there's some sort of permission problem or perhaps AV blocking some functionality. Maybe double check and make sure that AV isn't interfering by excluding MM folder. MM uses HUnspell which initially gave me a number of issues with various AV software until I switched to a different version. Perhaps AV is flagging that somehow or you might still have that older version around.

IAC, I'm at the end of my line with why this isn't working. I haven't heard this from anybody else, so I assume this is not a common problem.

**Comment by ArthurZ** [comment]
Appreciate your efforts Rick.
Unfortunately, no candle. 
Did exactly that in process discovered Chrome had a handle to Program Files/Markdown Monster.
Removed. Re-installed. A deep sigh.
Checked the AV pointers. Can't agree because I have ESET on my work machine, and ESET on my personal to which I installed the same (latest) MM and the SpellChecker works there.
The difference is Win 10 Home vs. Win 7 Pro.

**Comment by RickStrahl** [comment]
Hmmm... Can you just try to turn off ESET on the Win7 machine temporarily and see if that changes anything?

I don't think that Win7 should have any issues with MM. If there was a problem I'd suspect it to be affecting everything and not just the spell checker of all things. I'll see if I can set up a Win7 VM on Azure and see if anything comes up there.

**Comment by ArthurZ** [comment]
I did disabled it completely, does not help.

**Comment by RickStrahl** [comment]
So I set up an Azure Win7 VM and I was able to confirm the problem you describe - the spellchecker is not turning on. As a matter of fact a whole bunch of stuff is not working.

I believe this was broken when I added DPI awareness to the browser which didn't actually work with Windows 7. I was able to isolate that code and fix the problem, so now it works properly.

Thanks for your patience with this!

4f4a4f25bbc20506838cb4d3a9659235730a0c0c
